### PR TITLE
Backbone-no-x

### PR DESCRIPTION
**Pull request by shackbarth**
An investigation into what it would take to have our backbone layer talk to REST instead of websockets.

### CODE REVIEW

File: lib/enyo-x/source/views/workspace.js [hunk 1]
```diff
@@ -185,6 +185,7 @@ trailing:true, white:true, strict: false*/
       i;
 
     model[action]("change", ws.attributesChanged, ws);
+    model[action]("sync", ws.attributesChanged, ws);
```
Comment (shackbarth) [inline lib/enyo-x/source/views/workspace.js]: I really didn't want to have to touch the enyo layer but `sync` really is the way that backbone notifies listeners that it has gotten a response from the server.

File: lib/enyo-x/source/views/workspace.js [hunk 2]
```diff
@@ -566,6 +567,7 @@ trailing:true, white:true, strict: false*/
           id: this.value.id,
           done: options.modelChangeDone
         };
+      //options.patch = true;
```
Comment (shackbarth) [inline lib/enyo-x/source/views/workspace.js]: This is one way you could turn the edit request from a `PUT` to a `PATCH`.

**Comment by shackbarth** [comment]
Implemented for `Honorific`. It doesn't all work. See extensive comments at the top of `rest_model.js`.

File: lib/backbone-x/source/rest_model.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    },
+    //
+    // Describe the location of the REST endpoint
+    //
+    url: function () {
+      // let's keep the concept of recordType, but map the PascalCase to the
+      // the armadillo-case that our REST service expects
+      var recordType = this.recordType.substring(3);
+      var path = XT.String.decamelize(recordType).replace(/_/g, "-");
+      return XT.getOrganizationPath() + "/browser-api/v1/resources/" + path + "/" + this.id;
```
Comment (bendiy) [inline lib/backbone-x/source/rest_model.js]: TODO: Get this from the discovery document.

File: lib/backbone-x/source/rest_collection.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    //
+    // Describe the location of the REST endpoint
+    //
+    url: function () {
+      // let's keep the concept of recordType, but map the PascalCase to the
+      // the armadillo-case that our REST service expects
+      var recordType = this.model.prototype.recordType.substring(3);
+      var path = XT.String.decamelize(recordType).replace(/_/g, "-");
+      return XT.getOrganizationPath() + "/browser-api/v1/resources/" + path;
```
Comment (bendiy) [inline lib/backbone-x/source/rest_collection.js]: TODO: Get this from the discovery document.

**Comment by gilmoskowitz** [comment]
Merging in spite of CI failure because the destination is an experimental branch, not a pre-production working branch.

**Pull request merged**